### ISSUE TITLE
Using `expect` from built-in test context triggers `prefer-importing-vitest-globals`

### ISSUE THREAD

**Issue by gtbuchanan** [vitest-dev/eslint-plugin-vitest#724]
Using `expect` from the built-in test context ([as recommended by Vitest for concurrent tests](https://vitest.dev/guide/test-context.html#built-in-test-context)) triggers the new `prefer-importing-vitest-globals` rule.

```typescript
import { describe, it, test } from 'vitest';

describe.concurrent('local expect', () => {
  it('should not trigger lint rules', ({ expect }) => { // Triggers `prefer-importing-vitest-globals`
    expect(true).toBe(true);
  });
});
```

I assume this also occurs for any function that accepts the local `expect` as a parameter of the same name.
